gluten-ut/test/src/test/scala/org/apache/gluten/extension/GlutenExtensionRewriteRuleSuite.scala

```diff
@@ -62,4 +62,15 @@ class GlutenExtensionRewriteRuleSuite extends WholeStageTransformerSuite {
       }
     )
   }
+
+  test("GLUTEN-9279 - Not Pull out expression to avoid invalid reference binding") {
+    withTable("t") {
+      sql("CREATE TABLE t(f1 String, f2 String, f3 String, f4 String) USING CSV")
+      sql("INSERT INTO t values ('1', '2', '3', '4'), ('11' ,'22', '33', '4')")
+      val df = sql("""
+                     |SELECT SUM(f1) / COUNT(DISTINCT f2, f3) FROM t GROUP BY f4;
+                     |""".stripMargin)
+      assert(df.collect()(0).getDouble(0) == 6)
```

Comment:
You can use runQueryAndCompare to add ProjectExecTransformer and HashAggregateExecBaseTransformer checks to ensure offload.

Reply:
fixed